README.md

```diff
@@ -0,0 +1,684 @@
+  "can": "crud/update",
+  "iff": [
+     {                                    // ┐
+       "uri": "https://blog.example.com", // │
+       "status": "published"              // ├─ Caveat
+       "tag": "news",                     // │
+     },                                   // ┘
+     {                   // ┐
+       "tag": "breaking" // ├─ Caveat
+     }                   // ┘
+  ],
+  // ...
+}
+```
+
+The above MUST be interpreted as the set of capabilities below in the following table:
+
+| Subject               | Ability       | Caveat                                                                                                 |
+|-----------------------|---------------|--------------------------------------------------------------------------------------------------------|
+| `did:web:example.com` | `crud/update` | Posts at `https://blog.example.com` with the `published` status and tagged with `news` and `breaking`  |
+
+The `iff` caveat set MUST take the following shape: `[{}]`. The arrays represents logical `all` (chained `AND`s). To represent logical `any` / `OR`, issue another delegation with that attenuation. For instance, the following represents `{a: 1, b:2} AND {c: 3} AND {d: 4}`:
+
+``` js
+[
+  {       // ┐
+    a: 1, // ├─ Caveat ─┐
+    b: 2  // │          │
+  },      // ┘          ├─ AND ─┐
+  {       // ┐          │       │
+    c: 3  // ├─ Caveat ─┘       │
+  },      // ┘                  ├─ AND
+  {       // ┐                  │
+    d: 4  // ├─ Caveat ─────────┘
+  }       // ┘
+]
+```
+
+Expressing caveats in this standard way simplifies ad hoc extension at delegation time. As a concrete example, if a root UCAN caveat has a `tag` field but no (plural) `tags` field, it is still possible to express multiplicity by adding another `AND`ed caveat:
+
+``` js
+// Original Caveat
+[
+  {
+    "uri": "https://blog.example.com/",
+    "tag": "news"
+  }
+]
+
+// Attenuated Caveat
+[
+  {
+    "uri": "https://blog.example.com/",
+    "tag": "news"
+  }, 
+  // AND
+  {
+    "tag": "breaking"
+  }
+]
+```
+
+In this example, the original caveat had not accounted for there being multiple tags at runtime. The attenuated capability grants access to blog posts at `https://blog.example.com/` that are tagged with both `news` and `breaking` due to the `AND` in the predicate.
+
+This is also helpful if each object has a special meaning or sense:
+
+``` js
+[
+  {
+    "type": "path",
+    "segments": ["blog", "october"]
+  },
+  {
+    "type": "market",
+    "segments": ["manufacturing", "healthcare", "service", "technology"]
+  }
+]
+```
+
+Note that while adding whole objects is useful in many situations (as above), attenuation MAY also be achieved by adding fields to an object:
+
+``` js
+// Original Caveat
+[
+  {
+    "uri": "https://blog.example.com/",
+    "tag": "news"
+  }
+]
+
+// Attenuated Caveat
+[
+  {
+    "uri": "https://blog.example.com/",
+    "tag": "news",
+    "status": "draft" // New field
+  }
+]
+```
```

Comment:
Let me expand on my points around having proper operators as opposed to array of caveats that only executor will know how to merge / interpret. This is something I have been thinking for a long time and was planning to implement in user space specifically in ucanto RPC. Here is are couple of examples from our set of capabilities

```js
// delegate all store abilities without any constraints
{
   sub: "did:key:zSpace"
   can: "store/*",
}

// can re-delegate ☝️ by constraining to specific ability and limiting write size
{
   sub: "did:key:zSpace"
   can: "store/add",
   whr: {
      size: { "<=": 2222 }
   }
}

// can re-delegate as long as new constraints do not violate ☝️
{
   sub: "did:key:zSpace"
   can: "store/add",
   whr: {
      size: { "<=": 2000, ">": 1000 } // size is within given range
   }
}


// can re-delegate as long as new constraints do not violate ☝️
{
   sub: "did:key:zAlice",
   can: "store/add",
   whr: {
      archive: { "=": { "/": "bag...car" } },   // can only add specific CAR file
      size: { "=": 1234 }                               // which must have this exact size
   }
}
```

Here is the tags example from the above

```js
// Original Caveat
{
    "uri": { "=": "https://blog.example.com/" },
    "tag": { "⊇": ["news"] } // superset of the following tags
  }

// Attenuated Caveat (Must satisfy ☝️ caveats)
{
    "uri": { "=": "https://blog.example.com/" },
    "tag": { "⊇": ["news", "breaking"] } // superset of the following tags
  }


// Attenuated Cavet (Must satisfy ☝️ caveats)
{
    "uri": { "=": "https://blog.example.com/" },
    // can be super set of the specified tags but may not include other tags
    "tag": { "⊇": ["news", "breaking"], "⊄": ["blockchain"] }
}
```

We could also do something similar with arbitrary strings / paths

```js
// Delegates access to path as long as it starts with /public/ 
{
  "path": { "|=": "/public/" }
}

// Can attenuate as long as ☝️ caveat is satisfied
{
   "path": { "|=": "/public/photos/" } // as long as starts with /public/photos/
}

// Can attenuate as long as ☝️ caveat is satisfied
{
  "path": {
       // starts with /public/photos/
       "|=": "/public/photos/",
        // ends with ".jpeg"
        "=|": ".jpeg"
   } 
}

// Can attenuate as long as ☝️ caveat is satisfied
{
   "path": {
      "=": "/public/photos/profile.jpeg"
   }
}
```

None have OR operators even though in some case it may be handy, however that's ok as ORs can be achieved through separate delegations

Reply:
That's basically what we have, but we're not going to ship a bunch of UTF8. There will be a standard library to drive these mainly synatctically, but we need this to be extensible. WNFS has use cases that cannot be handled with the above, for example.

Reply:
Instead of this..

```json
// Original Caveat
{
    "uri": { "=": "https://blog.example.com/" },
    "tag": { "⊇": ["news"] } // superset of the following tags
  }

// Attenuated Caveat (Must satisfy ☝️ caveats)
{
    "uri": { "=": "https://blog.example.com/" },
    "tag": { "⊇": ["news", "breaking"] } // superset of the following tags
  }
```

...where you have to understand how to prove each caveat, we think it would be much easier to just have them as separate conditions. Using your notation:


```json
{
    {"field": "uri", "=": "https://blog.example.com/"},
    {"field": "tag", "include": "news" },
    {"field": "tag", "include": "breakling" }
  }
```

In this version, you can treat the array as an `all` predicate, and you don't have to understand the structure of each condition. At each delegation, it's just "did we add more conditions and leave the old ones as they were?"

Reply:
We also don't need `"="` in the caveats (now "conditions"), because that's what's precisely captured by the `args` field